File: src/main/java/controls/lambdas/Example3.java
package controls.lambdas;

import java.util.function.BiFunction;

public class Example3 {
    @SuppressWarnings("Convert2MethodRef")
    public void doIt() {
        System.out.println(withTwo("Hello", 1, (s, i) -> s.substring(i)));
    }

    public void doItWithMethodReference() {
        System.out.println(withTwo("Hello", 1, String::substring));
    }

    public String withTwo(String s, int i, BiFunction<String, Integer, String> f) {
        return f.apply(s, i);
    }
}

File: src/main/scala/examples/lambdas/Example3.scala
package examples.lambdas

  class Example3 {
    def doIt(): Unit = {
      val result = withTwo("Hello", 1, _.substring(_))
      println(result)
    }

    def withTwo(s: String, i: Int, f: (String, Int) => String): String =
      f.apply(s, i)
  }


